### ISSUE TITLE
recommended installation method of traffic fails in docker

### ISSUE THREAD

**Issue by hugocool** [xoolive/traffic#82]
Docker file:
```
FROM heroku/miniconda

ADD ./webapp /opt/webapp/
WORKDIR /opt/webapp

RUN conda install cartopy shapely

# Install dependencies
ADD ./webapp/requirements.txt /tmp/requirements.txt
RUN pip install -qr /tmp/requirements.txt

CMD gunicorn --bind 0.0.0.0:$PORT wsgi
```

This image will build and run fine for as long as traffic is not in the requirements.txt

If traffic is included in requirements .txt the following error is generated:

> Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-5v7q58/pandas/
> Error response from daemon: The command '/bin/sh -c pip install -qr /tmp/requirements.txt' returned a non-zero code: 1
> Failed to deploy '<unknown> Dockerfile: Dockerfile': Can't retrieve image ID from build stream
> 
> 

So which installation method is recommended in docker?
Since pip installing traffic fails without conda installing cartopy and shapely first.
I also tried using an apt builpack and running apt get libgeos and libproj before pip installing traffic, however now an installation error was thrown w.r.t numpy..

This package is tricky to install and breaks conventional environment managers!

**Comment by xoolive** [comment]
I never tried Dockerfile for traffic but it could be a good addition to share indeed. Did you have a look at the travis.yml config? It works like a charm in there (without conda).

I am busy these days but if you can’t manage to get it ready (PR welcome!) I can try myself later (maybe June?) after I recall how this whole docker world works...

**Comment by rmonstein** [comment]
I have the following Docker file that works well:

`FROM jupyter/pyspark-notebook:d4cbf2f80a2a

# the tag d4cbf2f80a2a has an docker container image with Spark 2.4.3

# copy miniconda files to image
COPY env_osn.yml env_osn.yml

# install zip
USER root
RUN apt-get update && apt-get install -y zip

# install nb_conda into the base python to allow to choose the environment in the jupyter notebook
USER jovyan
RUN conda install -y nb_conda

# install environment
USER jovyan
RUN conda env create -f env_osn.yml && conda clean -tipsy`

I use the jupyter/pyspark Docker file as a basis but I guess it should work in a similar way with a different basis. The  Conda file that is used to generate the environment from looks something like this:

`name: osn_env
channels:
  - conda-forge
  - defaults
dependencies:
  - python=3.7.7
  - numpy=1.18.1
  - pandas=1.0.3
  - cartopy
  - shapely
  - pip
  - nb_conda
  - pip:
    - traffic
    - keplergl
`
I hope that helps...

**Comment by xoolive** [comment]
fixed with #90 
See https://traffic-viz.github.io/docker.html (online in a minute)
